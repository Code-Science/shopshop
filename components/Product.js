import React, { useState } from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';
import styles from '../styles/components/Product.module.css';

function Product({ item }) {
  const [imageUrl, setUrl] = useState(
    item?.image1 ? item.image1 : '/images/bottomPic1.jpg'
  );

  return (
    <div className={styles.product}>
      <figure
        className={styles.figure}
        onMouseEnter={() => setUrl(item.image2)}
        onMouseLeave={() => setUrl(item.image1)}
      >
        <img src={imageUrl} alt="home banner" className={styles.img} />
      </figure>
      <div className={styles.details}>
        <p className={styles.name}>{item.product_name}</p>
        <p className={styles.price}>Â£{item.price}</p>
        <Link href={`/product/${item.id}`}>
          <a className={styles.productBtn}>Details</a>
        </Link>
      </div>
    </div>
  );
}

Product.propTypes = {
  item: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default Product;
