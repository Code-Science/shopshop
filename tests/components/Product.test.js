import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import Product from '../../components/Product';

configure({ adapter: new Adapter() });

describe('Product Component', () => {
  const mockItem = {
    id: 4,
    product_name: 'shirt',
    price: 80,
    image1: 'image1',
    image2: 'image2',
  };
  const product = shallow(<Product item={mockItem} />);

  it('expect to render product component', () => {
    expect(product.length).toEqual(1);
  });
  it('snapshot test', () => {
    expect(product).toMatchSnapshot();
  });
  it('change image on hover', () => {
    const image = product.find('figure');
    image.simulate('mouseenter'); // not working somehow
    expect(image.childAt(0).prop('src')).toEqual(mockItem.image1);
  });
});
