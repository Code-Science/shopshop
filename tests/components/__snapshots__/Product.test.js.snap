// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component snapshot test 1`] = `
<div
  className="product"
>
  <figure
    className="figure"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <img
      alt="home banner"
      className="img"
      src="image1"
    />
  </figure>
  <div
    className="details"
  >
    <p
      className="name"
    >
      shirt
    </p>
    <p
      className="price"
    >
      Â£
      80
    </p>
    <Link
      href="/product/4"
    >
      <a
        className="productBtn"
      >
        Details
      </a>
    </Link>
  </div>
</div>
`;
